test/user.test.js

```diff
@@ -1800,6 +1800,105 @@ describe('User', function() {
     });
   });
 
+  describe('Session validation after password reset', function() {
+    it('invalidates session when password is reset', function(done) {
+      var email = 'abcd@example.com';
+      var password = 'originalPass';
+      var user, tempToken;
+
+      async.series([
+        function createOriginalAccount(next) {
+          User.create({ email: email, password: password }, function(err, userInstance) {
+            if (err) return next(err);
+            user = userInstance;
+            next();
+          });
+        },
+        function(next) {
+          User.login({ email: email, password: password }, function(err, accessToken1) {
+            if (err) return next(err);
+            assert(accessToken1);
+            next();
+          });
+        },
+        function(next) {
+          User.login({ email: email, password: password }, function(err, accessToken2) {
+            if (err) return next(err);
+            assert(accessToken2);
+            next();
+          });
+        },
+        function(next) {
+          user.replaceAttributes({'email': email, password: 'newPass'}, function(err, user2) {
```

Comment:
How is `email` related to password reset? It should be enough to call `user.replaceAttribute('password', 'newPass', cb)`.

The same applies for other tests below.